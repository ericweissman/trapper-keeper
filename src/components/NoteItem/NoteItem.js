import React from 'react'

export const NoteItem = (props) => {
  const { handleItemChange, handleItemDelete, toggleComplete } = props;
  const { isCompleted, id, description } = props.item;

  const toggle = () => toggleComplete(id)
  const deleteItem = () => handleItemDelete(id)
  
  return (
    <div>
      {
        isCompleted ?
          <input checked type="checkbox" className="checkbox" onChange={toggle} /> :
          <input type="checkbox" className="checkbox" onChange={toggle} />
      }
      <input
        type="text"
        name={id}
        value={description}
        onChange={handleItemChange}
      />
      <button className="item-del" onClick={deleteItem}>x</button>
    </div>
  )
}

export default NoteItem