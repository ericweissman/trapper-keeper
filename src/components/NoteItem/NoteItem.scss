.noteitem-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  transition: 2s linear;
}

.item-input {
  margin-left: 10px;
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  width: 215px;
  height: 28px;
}

.checkbox {
  margin: 0px;
  transform: scale(2.3);
  padding: 0;
}

.item-del {
  height: 28px;
  font-family: $font-secondary;
  font-size: 23px;
  color: white;
  width: 45px;
  border-top-right-radius: 5px;
  outline: none;
  border-bottom-right-radius: 5px;
  border: 0px;
  background-color: $primary-dark;
      &:hover {
    transform: scale(1.01);
    background-color: $primary;
    transition: background-color .1s ease-out;
    }
}