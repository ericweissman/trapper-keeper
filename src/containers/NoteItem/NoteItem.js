import React, { Component } from 'react'

export class NoteItem extends Component {

  deleteItem = () => {
    this.props.handleItemDelete(this.props.item.id)
  } 

  render() {
    const { handleItemChange } = this.props;  
    return (
      <div>
        <input type="checkbox"/>
        <input
          type="text"
          name={this.props.item.id}
          value={this.props.item.description}
          onChange={handleItemChange}
        />
        <button onClick={this.deleteItem}>x</button>
      </div>
    )
  }
}

export default NoteItem