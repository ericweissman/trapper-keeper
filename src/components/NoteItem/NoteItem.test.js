import React from 'react'
import { NoteItem } from './NoteItem'
import { shallow } from 'enzyme'

describe('NoteItem', () => {
  let wrapper;
  const mockItem = {
    description: 'some item',
    id: 3,
    isCompleted: false,
    noteID: 10,
    timestamp: 99

  }

  const mockHandleItemChange = jest.fn()
  const mockHandleItemDelete = jest.fn()
  const mockToggleComplete = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <NoteItem
        item={mockItem}
        handleItemChange={mockHandleItemChange}
        handleItemDelete={mockHandleItemDelete}
        toggleComplete={mockToggleComplete}
      />
    )
  })

  it('should match the snapshot if item is unchecked', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if item is checked', () => {
    wrapper.setProps({ item: { isCompleted: true } })
  
    expect(wrapper).toMatchSnapshot()
  })

  it('should register a click on the checkbox', () => {
    const mockEvent = { target: { checked: true } }

    wrapper.find('.checkbox').simulate('change', mockEvent)
    expect(mockToggleComplete).toHaveBeenCalled()
  })

  it('should register a click on the delete button', () => {
    wrapper.find('.item-del').simulate('click')
    expect(mockHandleItemDelete).toHaveBeenCalled()
  })
})