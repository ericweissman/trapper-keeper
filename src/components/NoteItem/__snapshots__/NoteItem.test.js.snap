// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem should match the snapshot if item is checked 1`] = `
<div
  className="noteitem-container"
>
  <input
    checked={true}
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <input
    autocomplete="off"
    className="form-input item-input"
    onChange={[MockFunction]}
    placeholder="new item"
    type="text"
  />
  <button
    className="item-del"
    onClick={[Function]}
  >
    x
  </button>
</div>
`;

exports[`NoteItem should match the snapshot if item is unchecked 1`] = `
<div
  className="noteitem-container"
>
  <input
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <input
    autocomplete="off"
    className="form-input item-input"
    onChange={[MockFunction]}
    placeholder="new item"
    type="text"
  />
  <button
    className="item-del"
    onClick={[Function]}
  >
    x
  </button>
</div>
`;
